include/veriblock/blockchain/pop/fork_resolution.hpp

```diff
@@ -58,17 +58,33 @@ int getConsensusScoreFromRelativeBlockStartingAtZero(int64_t relativeBlock,
 }
 
 struct KeystoneContextList {
-  const std::vector<KeystoneContext>& ctx;
+  std::vector<KeystoneContext> ctx;
   const int keystoneInterval;
 
-  KeystoneContextList(const std::vector<KeystoneContext>& c, int keystoneInt)
-      : ctx{c}, keystoneInterval(keystoneInt) {}
+  KeystoneContextList(const std::vector<KeystoneContext>& c,
+                      int keystoneInt,
+                      int finalityDelay)
```

Comment:
maybe pass according config and take keystoneInterval and finalityDelay from there?